front/src/modules/timesStops/helpers/computeMargins.tsx

```diff
@@ -0,0 +1,94 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+import { mToMm, msToS } from 'utils/physics';
+
+import { formatDigitsAndUnit } from './utils';
+
+const emptyMargins = {
+  theoreticalMargin: '',
+  theoreticalMarginSeconds: '',
+  calculatedMarginSeconds: '',
+  diffMarginsSeconds: '',
+};
+
+function getTheoreticalMargin(selectedTrainSchedule: TrainScheduleResult, pathStepId: string) {
+  const pathStepIndex = selectedTrainSchedule.path.findIndex(
+    (pathStep) => pathStep.id === pathStepId
+  );
+  let theoreticalMarginIndex;
+  if (pathStepIndex === 0) {
+    theoreticalMarginIndex = 0;
+  } else {
```

Comment:
You can replace your loop findIndex by 2 tests, which will be more efficient and optimized
```suggestion
  if (selectedTrainSchedule.path.length === 0) return undefined;

  if (selectedTrainSchedule.path[0].id === pathStepId) {
    return selectedTrainSchedule.margins?.values[0];
  }
```

Reply:
great change, thanks